### ISSUE TITLE
Cache scan doesn't respect shared folders, nor can it be cancelled

### ISSUE THREAD

**Issue by GoogleCodeExporter** [0xfab-ri/universal-media-server#7]
```
What steps will reproduce the problem?
1. fresh install of UMS
2. select shared folders
3. tick enable the cache checkbox
4. tick hide cache folder (I suspect this step is not necessary)
5. hit save button
6. hit scan all shared folders button and confirm dialog (at this point UMS 
started scanning things not within my just selected shared folders)
7. hit stop scanning button.


What is the expected output? What do you see instead?
UMS should scan my selected shared folders. 
It was scanning the previously selected option (all folders)
UMS should stop scanning when i hit the stop button and allow my to continue 
using the GUI.
The interface locks up and I had to kill the process. I didn't test if 
streaming was affected as I had yet to finish configuration.

What version of the product are you using? On what operating system?
1.2.1 on Windows 7 64-bit

Please provide any additional information below.


```

Original issue reported on code.google.com by `mafus...@gmail.com` on 8 Jul 2012 at 11:00

Attachments:
- [debug.log](https://storage.googleapis.com/google-code-attachments/universal-media-server/issue-7/comment-0/debug.log)

**Comment by GoogleCodeExporter** [comment]
```
Same issue here, start scannig all the system can't be cancelled and program 
starts to hang and becomes unresposnive. I'm not linking the log cause there 
are no errors in it it just keep scanning with ffmpeg the files like in the bug 
reportee's log
```

Original comment by `ivan.g.o...@gmail.com` on 10 Aug 2012 at 6:59

**Comment by GoogleCodeExporter** [comment]
```
Scans can be cancelled as of 2.5.1 and we are working on the issue where it 
only scans the previous folder selection
```

Original comment by `subjunk` on 15 Mar 2013 at 11:20
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in the version after 2.5.1 (to be released)
```

Original comment by `subjunk` on 16 Mar 2013 at 5:26
- Changed state: **Fixed**
